src/modules/app-db/repositories/nft.repository.ts

```diff
@@ -218,17 +248,55 @@ export class NftRepository {
 
 
     for (const nftData of nfts) {
-      const { id, name, metadata = null, image } = nftData;
+      const { id, name } = nftData;
+      let { image } = nftData;
+      let { metadata = null } = nftData;
+
+      //If metadata contains prefix ipfs://ipfs/ or prefix no prefix, just hash add prefix https://flk-ipfs.xyz/ipfs/
+      if (metadata != null) {
+        if (metadata.startsWith("ipfs://ipfs/")) {

```

Comment:
To reduce duplication please wrap this to a function such as convertIpfsUrl(url).

Ideally nft repository should not do any fetching, this could be done if all preprocessing logic would be done in meta.service.ts.
Because NftInterface and CollectionInterface share the same data you could do that by introducing a function preprocessMetadata(item:NftInterface|CollectionInterface) in meta.service that would be called before assigning nfts or collections.